lib/models/nutrition/nutritional_plan.dart

```diff
@@ -226,15 +227,25 @@ class NutritionalPlan {
   /// returns diary entries
   /// deduped by the combination of amount and ingredient ID
   List<Log> get dedupDiaryEntries {
-    final out = <Log>[];
+    //Get unique and sort by frequency
+    const Duration recentWindow = Duration(days: -90);
+    var uniqueLogs = <String, Log>{};
     for (final log in diaryEntries) {
-      final found = out.firstWhereOrNull(
-        (e) => e.amount == log.amount && e.ingredientId == log.ingredientId,
-      );
-      if (found == null) {
-        out.add(log);
-      }
+      final int howMany = diaryEntries
+          .where((diaryEntries) =>
+              diaryEntries.ingredientId == log.ingredientId &&
+              diaryEntries.amount == log.amount &&
+              diaryEntries.datetime.isAfter(DateTime.now().add(recentWindow)))
+          .length;
```

Comment:
hmm we do (diaryEntries.length)^2 iterations here. could be a performance issue. i have a lot of diary entries.
perhaps we could iterate once and update a map of ingredientId->count records.

not sure what's the best way to compare the performance though, does dart come with benchmark tools?